src/frontend/js/widgets/UserLogin/index.tsx

```diff
@@ -73,10 +75,40 @@ const UserLogin = ({ profileUrls = {} }: UserLoginProps) => {
    */
   const { user, destroy, login, register } = useSession();
   const intl = useIntl();
+  const joanieUserAbilities = useJoanieUserAbilities();
+
+  const userMenuItems = useMemo(() => {
+    if (joanieUserAbilities) {
+      const unauthorizedUrlFilter = (urlItem: UserMenuItem) => {
+        if (urlItem.key === 'dashboard_teacher') {
+          return (
+            !!joanieUserAbilities &&
+            joanieUserAbilities.can(abilityActions.ACCESS_TEACHER_DASHBOARD)
+          );
+        }
+        return true;
+      };
+      return [
+        ...Object.entries(profileUrls)
+          .map(([key, { label, action }]) => ({
+            key,
+            label,
+            action: typeof action === 'string' ? bindUserDataToUrl(action, user) : action,
+          }))
+          .filter(unauthorizedUrlFilter),
+        {
+          key: 'logout',
+          label: intl.formatMessage(messages.logOut),
+          action: destroy,
+        },
+      ];
+    }
+    return [];
+  }, [user, joanieUserAbilities]);
 
   return (
     <div className="user-login">
-      {user === undefined ? (
+      {user === undefined || joanieUserAbilities === null ? (
```

Comment:
I still do not understand when joanieUserAbilities can be equal to null with the current logic.